src/main/java/org/tikv/txn/gc/GCWorker.java

```diff
@@ -0,0 +1,492 @@
+
+package org.tikv.txn.gc;
+
+import com.google.common.util.concurrent.ThreadFactoryBuilder;
+import com.google.protobuf.ByteString;
+import io.etcd.jetcd.ByteSequence;
+import io.etcd.jetcd.op.Op;
+import io.etcd.jetcd.options.PutOption;
+import org.apache.log4j.Logger;
+import org.tikv.common.PDClient;
+import org.tikv.common.TiConfiguration;
+import org.tikv.common.TiSession;
+import org.tikv.common.codec.KeyUtils;
+import org.tikv.common.exception.GCException;
+import org.tikv.common.exception.RegionException;
+import org.tikv.common.meta.TiTimestamp;
+import org.tikv.common.region.RegionManager;
+import org.tikv.common.region.RegionStoreClient;
+import org.tikv.common.region.RegionStoreClient.RegionStoreClientBuilder;
+import org.tikv.common.region.TiRegion;
+import org.tikv.common.util.BackOffFunction;
+import org.tikv.common.util.BackOffer;
+import org.tikv.common.util.ConcreteBackOffer;
+import org.tikv.common.util.FastByteComparisons;
+import org.tikv.kvproto.Kvrpcpb;
+import org.tikv.txn.Lock;
+
+import java.util.List;
+import java.util.concurrent.*;
+import java.util.stream.Collectors;
+
+public class GCWorker implements AutoCloseable {
+  private final String uuid;
+  private final long l_uuid;
+  private final PDClient pdClient;
+  private boolean gcIsRunning;
+  private long lastFinish;
+  private Logger logger = Logger.getLogger(this.getClass());
+  private final RegionStoreClientBuilder regionStoreClientBuilder;
+  private final ScheduledExecutorService scheduler = Executors.newSingleThreadScheduledExecutor(new ThreadFactoryBuilder().setDaemon(true).build());
+  private final ScheduledExecutorService gcWorkerScheduler = Executors.newSingleThreadScheduledExecutor(new ThreadFactoryBuilder().setDaemon(true).build());
+  private ExecutorService gcTaskThreadPool;
+  private ExecutorCompletionService<GCTask> gcTaskService;
+
+  private static final ByteString GC_LAST_RUN_TIME_KEY = ByteString.copyFromUtf8("tikv_gc_last_run_time");
+  private static final ByteString GC_RUN_INTERVAL_KEY = ByteString.copyFromUtf8("tikv_gc_run_interval");
+  private static final ByteString GC_LIFE_TIME_KEY = ByteString.copyFromUtf8("tikv_gc_life_time");
+  private static final ByteString GC_SAFE_POINT_KEY = ByteString.copyFromUtf8("tikv_gc_safe_point");
+  private static final ByteString GC_CONCURRENCY_KEY = ByteString.copyFromUtf8("tikv_gc_concurrency");
+  private static final ByteString GC_LEADER_UUID_KEY = ByteString.copyFromUtf8("tikv_gc_leader_uuid");
+  private static final ByteString GC_LEADER_LEASE_KEY = ByteString.copyFromUtf8("tikv_gc_leader_lease");
+  public static final int gcScanLockLimit = 1024;
+  private final int gcDefaultRunInterval;// = 6 * 1000
+  private final int gcDefaultLifeTime;// = 2 * 1000;
+  private final int gcDefaultConcurrency;
+  private static final int gcMinConcurrency = 1;
+  private static final int gcMaxConcurrency = 128;
+  private final int gcWaitTime;// = 3 * 1000;
+  private final int gcWorkerLease;// = 2 * 1000;
```

Comment:
It seems these comments is not consistent with the default value, which is 2 * 60 * 1000 milliseconds?

Reply:
yeah, i will delete them